src/Microsoft.AspNetCore.Authorization/DefaultAuthorizationService.cs

```diff
@@ -51,16 +52,45 @@ public DefaultAuthorizationService(IAuthorizationPolicyProvider policyProvider,
 
             if (authContext.HasSucceeded)
             {
-                _logger.UserAuthorizationSucceeded(user?.Identity?.Name);
+                _logger.UserAuthorizationSucceeded(GetUserNameForLogging(user));
                 return true;
             }
             else
             {
-                _logger.UserAuthorizationFailed(user?.Identity?.Name);
+                _logger.UserAuthorizationFailed(GetUserNameForLogging(user));
                 return false;
             }
         }
 
+        private string GetUserNameForLogging(ClaimsPrincipal user)
+        {
+            var identity = user?.Identity;
+            if (identity != null)
+            {
+                var name = identity.Name;
+                if (name != null)
+                {
+                    return name;
+                }
+                return GetClaimValue(identity, "sub")
+                    ?? GetClaimValue(identity, ClaimTypes.Name)
+                    ?? GetClaimValue(identity, ClaimTypes.NameIdentifier)
+                    ?? GetClaimValue(identity, ClaimTypes.WindowsAccountName)
```

Comment:
@PinpointTownes Unfortunately not every provider sets it. Because, meh. Fine, name, sub, nameidentifier,